include/core/evaporation_model.h

```diff
@@ -0,0 +1,759 @@
+ * it, and/or modify it under the terms of the GNU Lesser General
+ * Public License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ * The full text of the license can be found in the file LICENSE at
+ * the top level of the Lethe distribution.
+ *
+ * ---------------------------------------------------------------------
+ */
+#ifndef lethe_evaporation_model_h
+#define lethe_evaporation_model_h
+
+#include <core/parameters.h>
+#include <core/physical_property_model.h>
+
+using namespace dealii;
+
+/**
+ * @brief EvaporationModel. Abstract class that allows to calculate the
+ * evaporation mass, heat and momentum fluxes. This model computes all the terms
+ * required for the assembly of the governing equations, i.e., Navier-Stokes
+ * momentum and energy, when evaporation is considered. The former terms are
+ * the momentum flux (i.e., pressure) acting on the evaporative front in the
+ * normal direction, the heat flux, acting at the same location as a cooling
+ * term, and the fluxes' jacobian, if required.
+ *
+ * The EvaporativeModel is the parent class, and its children are specific
+ * types of evaporative models, e.g., constant, temperature dependent. The
+ * selected model is instantiated in the assemblers' constructor
+ * (NavierStokesVOFAssemblerEvaporation and/or
+ * HeatTransferAssemblerVOFEvaporation), using the model_cast method according
+ * to the evaporation model parameters.
+ *
+ * The assemblers call the required method of the EvaporationModel for either
+ * the assembly of the rhs or matrix terms. Hence, no matter the selected
+ * evaporation model, the same assembler is called, avoiding repetitions of
+ * the assembler classes.
+ *
+ */
+class EvaporationModel
+{
+public:
+  EvaporationModel()
+  {}
+
+  /**
+   * @brief Instantiates and returns a pointer to an EvaporationModel
+   * object by casting it to the proper child class.
+   *
+   * @param evaporation_parameters Evaporation model parameters.
+   */
+  static std::shared_ptr<EvaporationModel>
+  model_cast(const Parameters::Evaporation &evaporation_parameters);
+
+  /**
+   * @brief Returns true if the EvaporationModel depends on a field, false if not.
+   * @param id Identifier of the field for which the function is called.
+   */
+  inline bool
+  depends_on(const field &id)
+  {
+    return model_depends_on[id];
+  }
+
+  /**
+   * @brief mass_flux Calculates the value of the evaporation mass flux.
+   * @param field_values Map storing the values of the various fields on which the
+   * mass flux may depend.
+   * @return Value of the flux calculated with the field_values.
+   */
+  virtual double
+  mass_flux(const std::map<field, double> &field_values) = 0;
+
+  /**
+   * @brief mass_flux Calculates the values of the evaporation mass flux
+   * for multiple points.
+   * @param field_vectors Map storing the vectors of the values for each fields
```

Comment:
You can also check for other instances of this, since it it is a recurring sentence :)
```suggestion
   * @param field_vectors Map storing the vectors of the values for each field
```